### PR TITLE
Fix #8114: Support CoW Swap 🐮 orders in Safer Sign

### LINKED ISSUE #8114: Safe Sign support for CoW 🐮 orders
### Description: 

Requires Desktop issue: https://github.com/brave/brave-browser/issues/33093

Core has the parsing work done, we should display these requests in our existing Safe Sign UI previously designed for 0x swaps.

Also should investigate using `GetEthTokenSymbol` / `GetEthTokenDecimals` for unknown tokens, currently we are only using `GetTokenInfo` from Asset Ratio Service but it only supports Mainnet.

When finding non-nil `EthSignTypedDataMeta? meta;`, we should display "Free" for the network fee as it's a sign request not a transaction.

cc @onyb

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Updates to use Safer Sign UI for displaying CoW swap orders.
- CoW Swap orders when swapping from ETH (native token) will create a `ETHSwap` transaction. For all other tokens, a sign message request is created (no gas fees!).
- A large part of this PR is refactoring UI components for re-use.
    - `SwapTransactionConfirmationView` was refactored into `SaferSignTransactionView` for re-use and removes dependency on `ParsedTransaction` for display. This was the safer sign UI for ethereum swap transactions.
        - `SaferSignTransactionContainerView` wraps this view for Safer Sign swap transactions.
        - `SaferSignMessageRequestContainerView` wraps this view for Safer Sign message requests (CoW Swap orders).
    - `SignMessageRequestView` was refactored to extract the content (message/domain display + consecutive new line & unknown character warnings). The message/domain & warning can now be displayed separately using `SignMessageRequestContentView`
        - `SaferSignMessageRequestContainerView` re-uses this `SignMessageRequestContentView` for the detail view, so we get the same display as sign message requests.
- A `SignMessageRequestStore` was created for `SignMessageRequestContainerView`. It's used to fetch details for CoW swap orders from core apis.

This pull request fixes #8114

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

Direct link to CoW Swap for WETH token -> COW token on Goerli Test Network (I used Goerli, any supported network sufficient).
https://swap.cow.fi/#/5/swap/WETH/COW

1. Test a CoW Swap trade using any non-native token.
2. Verify:
    1. Safer Sign Swap UI is shown. 
    2. Tapping `Details` shows the sign transaction details. 
    3. Tapping link button beside tokens links out to explorer website for the token.
    4. Gas fee is free!
3. Test a CoW Swap trade with a custom recipient (tap the gear, enable Custom Recipient toggle).
4. Verify:
    1. Safer Sign Swap UI is shown
    2. Verify custom recipient is listed in the to token section (tap & hold to see full address).
    3. Links + gas fee verified above in Step 2.
5. Test a CoW Swap trade using native token (ex. ETH on Goerli).
    - ⚠️ These trades cost gas, since they are real transactions. Also, we do not have the ability to display custom recipients in this case, for the time being.
6. Verify Safer Sign Swap UI is shown as expected (similar to Swap in native Brave Wallet). Gas Fee shouldn't be 0, as this is a transaction instead of a signature request.
7. Test a CoW swap with unknown tokens. If you used COW token for any of the above, this has been verified by the COW symbol displaying. Any token not in search assets should suffice!

## Screenshots:

CoW Swap | CoW Swap - details | _ 
--|--|--
CoW Swap - custom recipient | CoW Swap - custom recipient address | CoW Swap Details - custom recipient address

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/Crypto/Stores/CryptoStore.swift [hunk 1]
```diff
@@ -537,6 +537,30 @@ public class CryptoStore: ObservableObject, WalletObserverStore {
     }
   }
   
+  private var signMessageRequestStore: SignMessageRequestStore?
+  func signMessageRequestStore(for requests: [BraveWallet.SignMessageRequest]) -> SignMessageRequestStore {
+    if let store = signMessageRequestStore {
+      DispatchQueue.main.async { // don't update in view body computation
+        store.requests = requests
```
Comment (nuo-xu) [inline Sources/BraveWallet/Crypto/Stores/CryptoStore.swift:544]: so does `requests` need to be `@Published`?

File: Sources/BraveWallet/Crypto/Transaction Confirmations/SaferSignTransactionView.swift [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  /// The network the transaction belongs to
+  let network: BraveWallet.NetworkInfo?
+  
+  /// The address of the account making the swap
+  let fromAddress: String?
+  /// The name of the account
+  let namedFromAddress: String?
+  
+  /// The address of the recipient (applicable to CoW Swap)
+  let receiverAddress: String?
```
Comment (nuo-xu) [inline Sources/BraveWallet/Crypto/Transaction Confirmations/SaferSignTransactionView.swift:22]: if you agree, i think we should always show the receive wallet address/account

File: Sources/BraveWallet/Crypto/Stores/CryptoStore.swift (hunk 1 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/CryptoStore.swift:544]: Yes for when you have multiple requests and sign/cancel one, we need to update the `1 of x` UI

File: Sources/BraveWallet/Crypto/Transaction Confirmations/SaferSignTransactionView.swift (hunk 2 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/SaferSignTransactionView.swift:22]: I think fine to show for both, I'll update 👍

Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Transaction Confirmations/SaferSignTransactionView.swift:22]: Updated in [95d9bf9](https://github.com/brave/brave-ios/pull/8533/commits/95d9bf93899831106daad79bca950b1d13e83faf)

**Review by nuo-xu — APPROVED** [review]

**Comment by kdenhartog** [comment]
> These trades cost gas, since they are real transactions. Also, we do not have the ability to display custom recipients in this case, for the time being.

What do we display in this case? I'd assume the details are proper and we could see it there,but I don't expect many users will think to do that. I could see this being used as a tool to drain wallets.

**Comment by StephenHeaps** [comment]
> > These trades cost gas, since they are real transactions. Also, we do not have the ability to display custom recipients in this case, for the time being.
> 
> What do we display in this case? I'd assume the details are proper and we could see it there,but I don't expect many users will think to do that. I could see this being used as a tool to drain wallets.

In this case they are provided to us from BraveCore as `TransactionInfo` models (instead of `SignMessageRequest` models) display using the existing transaction confirmation (not sign message panel), with gas fees:

https://github.com/brave/brave-ios/assets/5314553/ac66f64f-3297-462c-990f-7d2826cb238a

**Comment by kdenhartog** [comment]
@StephenHeaps sorry, I misquoted what I was originally trying to comment on. I was referencing the custom recipients. I see on desktop as well as iOS that we'll want to fix. I'm opening an issue to get this fixed on Desktop. Do we need to do something to get this fixed on iOS in this PR or would it be easier to ship it in a follow up PR?

**Comment by StephenHeaps** [comment]
> @StephenHeaps sorry, I misquoted what I was originally trying to comment on. I was referencing the custom recipients. I see on desktop as well as iOS that we'll want to fix. I'm opening an issue to get this fixed on Desktop. Do we need to do something to get this fixed on iOS in this PR or would it be easier to ship it in a follow up PR?

In this PR, iOS supports displaying CoW Swap custom recipients for safer sign signature requests (gas-less / free network fee, shown with `Sign` button), but we don't have ability to parse out the recipient in the `ETHSwap` transaction type yet (requires gas, shown with `Confirm` button).

**Review by kdenhartog — APPROVED** [review]
LGTM the one concern I have requires core changes happening in https://github.com/brave/brave-browser/issues/34829 and then we can update that in iOS separately as well. Thanks @StephenHeaps for pointing out the custom recipients so I spent some time investigating it further.

**Pull request merged**